plugins/chat/app/services/chat/update_user_last_read.rb

```diff
@@ -68,6 +71,10 @@ def mark_associated_mentions_as_read(active_membership:, message:, **)
       )
     end
 
+    def update_channel_last_viewed_at(active_membership:, **)
+      active_membership.update!(last_viewed_at: Time.zone.now)
```

Comment:
We are already doing `active_membership.update!` in `def update_last_read_message_id`. We should try to group these together.

We also need to be careful about naming here, we call it `update_channel_last_viewed_at` and it's incorrect as we update the channel.

We need to come up with a step name for both... something like `update_membership_state` or whatever you think is appropriate.

Reply:
Yep fair enough